jetty-io/src/main/java/org/eclipse/jetty/io/ssl/SslConnection.java

```diff
@@ -187,11 +191,18 @@ public SslConnection(ByteBufferPool byteBufferPool, Executor executor, EndPoint
 
     public SslConnection(ByteBufferPool byteBufferPool, Executor executor, EndPoint endPoint, SSLEngine sslEngine,
                          boolean useDirectBuffersForEncryption, boolean useDirectBuffersForDecryption)
+    {
+        this(new AdapterMemoryPool(byteBufferPool), byteBufferPool, executor, endPoint, sslEngine, useDirectBuffersForEncryption, useDirectBuffersForDecryption);
```

Comment:
This should be findOrAdapt

Reply:
With an extra null check in the implementation, yes, that can be done. I'm okay with this idea as it would then be possible to make `AdapterMemoryPool` private as you pointed out below.